### PR TITLE
feat(serve): smart-rename structured view sessions from the first message

### LINKED ISSUE #2026: Smart rename of session
```
@Nathan Brake I'm coming from the Conductor world to AoE and one thing I miss from conductor is it would use the LLM to name the session/worktree on first message.    The (random civ) thing isn't quite as good.

Its was nice when the session name gave context about what you were working on in it
```
Now that we merged the work to tie the worktree and name together by default, this is nicely acheivable

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the implementation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

Structured view (ACP) sessions start with a random civilization name (`Vikings`), which carries no context about what the session is for. This adds an opt-out "smart rename": on the first message in a structured view session, AoE runs the session's own agent once in non-interactive one-shot mode (`claude -p`, `codex exec`, `opencode run`, `gemini -p`) to produce a short title and renames the session, the way Conductor/cmux do.

It is title only. The live ACP worker holds the worktree as its working directory, so the directory cannot be moved while the session runs (the same constraint a manual rename of a running tied session hits); the visible session title is what gains meaning. It runs only while the session still carries its auto-generated name, so a session you named yourself is never touched, and it is fully best-effort: a failed, timed-out, or off-format one-shot leaves the generated name and never blocks or fails your prompt.

Design notes:
- Whether a name is "still default" is detected statelessly against the civilization list (plus the Roman-numeral / timestamp suffixes the generator can emit), so there is no new persisted field and no migration. The gate is re-checked under the per-session lock before the title is written, so a manual rename (or a deletion) landing during the one-shot call always wins.
- One-shot support is a per-agent `AgentDef.oneshot_flag`; the four agents above are wired and verified against their CLIs. Agents with no one-shot mode, sandboxed sessions (a host one-shot lacks the container's auth), and command-overridden agents keep the generated name.
- User text is passed as a lone argv element (never through a shell), truncated and NUL-stripped; agent output is strictly sanitized (ANSI stripped, refusals / over-long / echoed responses rejected). An in-flight guard prevents a burst of rapid first prompts from spawning concurrent renames.

The naming knob discussed in the issue (`session.smart_rename`, default on) is the single-source schema setting, so it appears in both the TUI and the web settings automatically.

The sidebar also shows where each session stands: a dim sparkle `Auto-name` chip marks a still-default-named eligible session (it will be named on its first message), and an amber pulsing `Naming…` chip shows while the one-shot title call is in flight. The chips clear once renamed or when the session is not eligible. The state is computed server-side in `list_sessions` from the same `check_eligible` predicate that gates the rename, so the indicator cannot drift from the behavior.

Closes #2026

## PR Type

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## How to test

- [ ] Start the web dashboard, create a new structured view claude session with no title (it gets a civilization name like `Vikings`); the sidebar shows a sparkle `Auto-name` chip.
- [ ] Send a first message such as "help me add rate limiting to the login endpoint"; a `Naming…` chip appears briefly, then within ~12s the session title changes to a short summary and the chip clears. The worktree directory is unchanged.
- [ ] Rename the session manually first, then send a first message: the title is not auto-changed and no chip shows.
- [ ] Set `session.smart_rename = false` (or toggle "Smart Session Rename" off in Settings), create a session, send a message: the civilization name stays and no chip shows.
- [ ] Create a sandboxed structured view session and send a message: the civilization name stays (no error in the prompt flow).

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / structured view (Playwright user story)**
- [x] Added or updated a Playwright spec under `web/tests/` and updated `web/tests/coverage-matrix.json`

**TUI / CLI (e2e test)**
- [x] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle

## AI Usage

- [ ] No AI was used
- [x] AI was used

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires `/aoe-implement` skill)

**Any Additional AI Details you'd like to share:**
Investigation, plan, and implementation were drafted by Claude under my direction, with a multi-model `/debate` shaping the design. I made the design calls (title-only, stateless default-name detection, per-agent one-shot, the sidebar indicator UX, edge-case handling), reviewed each commit, verified the one-shot tokens against the real CLIs, and ran the manual test steps.

- [x] I am an AI Agent filling out this form (check box if true)

---
<a href="https://app.blacksmith.sh/agent-of-empires/codesmith/agent-of-empires/pr/2201"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1784194819&installation_id=139138837&pr_number=2201&repository=agent-of-empires%2Fagent-of-empires&return_to=https%3A%2F%2Fgithub.com%2Fagent-of-empires%2Fagent-of-empires%2Fpull%2F2201&signature=4594f340eed2b132bcca0608bf03243dcb6c0e01813be6d2cf1f296762f6827d"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

## What Changed

Structured-view (ACP) sessions now automatically replace the initial random “civilization-style” name with a **meaningful short title** derived from what the user says in the **first message**. The auto-rename only triggers while the session is still using its **default generated name**, so **manual titles are never overwritten**.

## What Was Added / Updated

- **Smart rename engine (new `smart_rename` module)**
  - Detects whether the current session title is still the default civilization-based name (including common suffix patterns like Roman numerals / timestamps).
  - Generates a candidate title by running the **session’s own agent** in **one-shot** mode on the first message.
  - **Best-effort only**: failures, timeouts, or malformed output never block prompts; if it can’t produce a valid title, it’s simply left as-is.
  - **Runs once per session** (no repeated attempts on later prompts).
  - Avoids concurrent renames using an in-flight guard, and prevents duplicate work via “already attempted” tracking.
  - Updates only the **displayed title** (the worktree directory isn’t moved).

- **Agent one-shot support**
  - Introduces a `oneshot_flag` per agent definition so only agents capable of safe one-shot invocation are used for naming.
  - Non-supported / sandboxed / command-overridden cases are skipped.

- **Backend/API + session state**
  - Session list responses now include a structured smart-rename lifecycle state (`inactive`, `pending`, `running`) so the UI can show accurate progress.

- **Web UI indicators**
  - Workspace sidebar adds chips to reflect rename status:
    - **“Auto-name”** when eligible and waiting
    - **“Naming…”** while title generation is running
  - Adds/updates tests (unit + Playwright/Vitest) covering the chip behavior and the new settings toggle.

- **User configuration**
  - Adds a `session.smart_rename` setting (default **enabled**) exposed in the UI as **“Smart Session Rename”**.
  - Updated docs describe how and when smart renaming occurs and when it’s skipped.

## Benefits

- Sessions become **immediately understandable**, since titles reflect the actual work described in the first message.
- The behavior is **safe and non-disruptive**: it never blocks user prompts and never overwrites user-provided titles.
- Clear UI feedback makes it obvious when auto-naming is queued or actively generating.

## Potential Areas for Further Enhancement

- If future agent support becomes available, expand which agents can participate in one-shot naming.
- Consider adding an optional “preview before apply” workflow for users who want visibility into the generated title before it replaces the default.

### CODE REVIEW

File: src/session/civilizations.rs [hunk 1]
```diff
@@ -109,6 +109,28 @@ pub fn generate_random_title(existing_titles: &[&str]) -> String {
...
+pub fn is_default_civ_name(title: &str) -> bool {
+    let t = title.trim();
+    if CIVILIZATIONS.contains(&t) {
+        return true;
+    }
+    if let Some((base, suffix)) = t.rsplit_once(' ') {
+        if CIVILIZATIONS.contains(&base) && !suffix.is_empty() {
+            let is_roman = suffix.chars().all(|c| "IVXLCDM".contains(c));
+            let is_timestamp = suffix.chars().all(|c| c.is_ascii_digit());
+            return is_roman || is_timestamp;
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/session/civilizations.rs:132]: Good catch. Tightened to match only what `generate_random_title` actually emits: a Roman numeral in 2..=1000 and a timestamp of >=9 digits, with tests asserting `Britons IM` and `Vikings 2` are rejected. Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/2201/commits/a0773758

File: src/session/smart_rename.rs [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+            return;
+        };
+
+        let Some(raw) = run_oneshot(&argv, &project_path).await else {
+            return;
+        };
+        let Some(new_title) = sanitize_title(&raw, &first_message) else {
+            tracing::debug!(target: "smart_rename", session = %session_id, "skip: agent output not a usable title");
+            return;
+        };
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/session/smart_rename.rs:353]: Agreed. Added a per-session attempted set on AppState, marked before `run_oneshot`, so a session gets exactly one one-shot per daemon lifetime; the `pending` chip is gated on it too. Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/2201/commits/ac295909

**Review by njbrake — APPROVED** [review]
